util.js

```diff
@@ -0,0 +1,302 @@
+        },
+    };
+    
+    await execInVenv('tutor', ['config', 'printroot'], tutor_root_options);
+    tutorRoot = tutorRoot.trim();
+  
+    // Validate tutor root path
+    if (!tutorRoot) {
+      throw new Error(`The tutor root path does not exist.`);
+    }
+  
+    return tutorRoot;
+}
+
+/**
+ * Installs themes for Open edX from a given GitHub repository and branch.
+ * @async
+ * @param {string} themeRepository - The GitHub URL of the theme repository.
+ * @param {string} themeBranch - The branch to clone from the theme repository.
+ * @param {string} tutorRoot - The root directory where Tutor is installed.
+ * @param {object} options - Additional options for the `exec` method.
+ * 
+ * @throws Will throw an error if the installation process fails.
+ * 
+ * @example -> await installThemes('https://github.com/example/repo.git', 'main', '/path/to/tutor/root', options);
+ */
+async function installThemes(themeRepository, themeBranch, tutorRoot, options) {
+    if (!themeRepository || themeRepository === 'false' || !themeBranch || !tutorRoot) {
+      core.info('Skipping theme installation due to insufficient parameters.');
+      return;
+    }
+  
+    const themesPath = `${tutorRoot}/env/build/openedx/themes/`;
+  
+    try {
+        core.info(`Starting theme installation from repository: ${themeRepository} and branch: ${themeBranch}`);
+      
+        // Clone the repository and check if it was successful
+        const cloneResult = await exec.exec('git', ['clone', '-b', themeBranch, themeRepository], options);
+        if (cloneResult !== 0) {
+          throw new Error("Failed to clone the theme repository.");
+        }
+        
+        // Check if the destination directories exist
+        if (!fs.existsSync(themesPath)) {
+          throw new Error(`Themes directory ${themesPath} doesn't exist.`);
+        }
+      
+        moveAll('openedx-themes/edx-platform/', themesPath);
+      
+        core.info('Themes installation completed successfully.');
+      } catch (error) {
+        throw new Error(`Failed to install themes. Error: ${error.message}`);
+      }
+}
+
+
+/**
+ * Clones a GitHub repository to a specified directory.
+ * @async
+ * @param {string} repository - The name of the repository to clone.
+ * @param {string|null} branch - The specific branch to clone. If null, the default branch will be cloned.
+ * @param {string} targetDir - The directory where the repository should be cloned.
+ * @param {Object} options - Additional options passed to the exec command.
+ * @param {string} ghAccessToken - GitHub access token for authentication.
+ * @throws Will throw an error if the execution fails.
+ * @example -> await gitClone('myRepo', 'main', './targetDir', {}, 'ghAccessToken');
+ */
+async function gitClone(repositoryName, branch, targetDir, options, ghAccessToken="" ) {
+    try {
+        const branchOption = branch ? ['-b', branch] : [];
+
+        const authPart = ghAccessToken ? `${ghAccessToken}@` : '';
```

Comment:
Unnecesarry blank line. @Serafin-dev 
```suggestion
        const branchOption = branch ? ['-b', branch] : [];
        const authPart = ghAccessToken ? `${ghAccessToken}@` : '';
```